src/Sentry/Internal/Extensions/JsonExtensions.cs

```diff
@@ -159,34 +159,35 @@ public static void Deconstruct(this JsonProperty jsonProperty, out string name,
             IDiagnosticLogger? logger,
             bool includeNullValues = true)
         {
-            if (dic is not null)
+            if (dic is null)
             {
-                writer.WriteStartObject();
+                writer.WriteNullValue();
+                return;
+            }
+
+            writer.WriteStartObject();
+
+            var ordered = dic.OrderBy(_ => _.Key, StringComparer.Ordinal);
```

Comment:
We probably have some `OrderBy` or `Sort` in existing tests that can be removed now.